pkg/operation/botanist/monitoring.go

```diff
@@ -60,7 +61,26 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		usersDashboards     = strings.Builder{}
 	)
 
-	// Find extensions provider-specific monitoring configuration
+	// Fetch component-specific monitoring configuration
+	for _, component := range []component.MonitoringComponent{
+		b.Shoot.Components.ControlPlane.KubeScheduler,
+	} {
+		scrapeConfig, err := component.ScrapeConfig()
+		if err != nil {
+			return err
+		}
+		scrapeConfigs.WriteString(fmt.Sprintln(scrapeConfig))
```

Comment:
A second thought about `scrapeConfigs`:
Maybe it would be better to use `[]string` for collecting the scrape configs, so implementors of `MonitoringComponent` don't have to now about the correct YAML-indent which must be used. Instead we should iterate over the collected scrape configs in the template and add the array item marker (`- `) and the right indentation, so it can be handled centrally, instead of handling it in all components.